### PR TITLE
Updated Sandbox to represent a single SITL session (fixes #123 and fixes #115 and fixes #74 and fixes #65)

### LINKED ISSUE #65: Wrap Sandbox.provision inside a context manager

### LINKED ISSUE #74: Lift "schemas" instance variable into a class variable for System

### LINKED ISSUE #115: Remove dependency from System to Sandbox

### LINKED ISSUE #123: Sandbox should provide access to a given Docker container
I.e., creating a sandbox for a system shouldn't create a new Docker container. Instead, the sandbox should accept a given Docker container and should launch an SITL instance inside that container.

### PR DESCRIPTION
**Pull request by ChrisTimperley**

### CODE REVIEW

File: houston/sandbox.py [hunk 1]
```diff
@@ -26,193 +25,152 @@ class Sandbox(object):
...
+        return outcome
+
+
+    def run(self, commands: Sequence[Command]) -> MissionOutcome:
+        """
+        Executes a mission, represented as a sequence of commands, and
+        returns a description of the outcome.
+        """
+        config = self.configuration
+        env = mission.environment
```
Comment (afsafzal) [inline houston/sandbox.py]: mission?

File: houston/sandbox.py [hunk 2]
```diff
@@ -26,193 +25,152 @@ class Sandbox(object):
...
         """
-        A flag indicating whether or not this sandbox is alive.
+        The BugZoo container underlying this sandbox.
         """
-        # FIXME should also check that container is alive via BugZoo API
-        return self.__container is not None
+        return self.__container
 
-    def _start(self, mission: Mission) -> None:
+    def _start(self) -> None:
```
Comment (afsafzal) [inline houston/sandbox.py]: Where is this called? I mean who instructs the sandbox to start the SITL?

File: houston/sandbox.py (hunk 1 above)
Comment (ChrisTimperley) [reply houston/sandbox.py]: Fixed.

File: houston/sandbox.py (hunk 2 above)
Comment (ChrisTimperley) [reply houston/sandbox.py]: `_start` and `_stop` are called from outside of the sandbox instance. There are two ways that those methods will now be called:

1. a static method of `Sandbox` tentatively called `build` will implement a context manager that launches a sandbox inside a given Docker container and automatically starts and stops the SITL as the context is entered and exited.
   ```
   with Sandbox.build(bz, container, state, env, config) as sandbox:
      pass
   ```

2. the caller manually constructs a `Sandbox` instance and invokes `_start` and `_stop` at the appropriate time. This is useful for cases where context managers are clunky or inappropriate.

**Pull request merged**